Confidence/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -77,10 +80,23 @@ class Confidence internal constructor(
         return flagResolver.resolve(flags, getContext())
     }
 
-    suspend fun awaitReconciliation() {
-        if (currentFetchJob != null) {
-            currentFetchJob?.join()
-            activate()
+    suspend fun awaitReconciliation(timeoutMillis: Long = 5000) {
```

Comment:
Does this support multiple concurrent calls? My concern is:
`awaitReconciliation (1)`
`awaitReconciliation (2)`
`putContext` -> this unlocks only (1)